controllers/openstackprovisionserver_controller.go

```diff
@@ -204,49 +216,71 @@ func (r *OpenStackProvisionServerReconciler) Reconcile(ctx context.Context, req
 		curURL, err := url.Parse(instance.Status.LocalImageURL)
 
 		if err != nil {
-			r.Log.Error(err, "Failed to parse existing LocalImageURL for OpenStackProvisionServer %s: %s", instance.Name, instance.Status.LocalImageURL)
+			cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.ProvisionServerCondTypeError)
+			cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.OpenStackProvisionServerCondReasonLocalImageURLParseError)
+			cond.Message = fmt.Sprintf("Failed to parse existing LocalImageURL: %s", instance.Status.LocalImageURL)
+			err = common.WrapErrorForObject(cond.Message, instance, err)
+
 			return ctrl.Result{}, err
 		}
 
 		// If the current LocalImageURL is empty, or its embedded IP does not equal the ProvisionIP, the update the LocalImageURL
 		if instance.Status.LocalImageURL == "" || curURL.Hostname() != instance.Status.ProvisionIP {
 			// Update status with LocalImageURL, given ProvisionIP status value
 			instance.Status.LocalImageURL = r.getLocalImageURL(instance)
-			err = r.Client.Status().Update(context.TODO(), instance)
-
-			if err != nil {
-				r.Log.Error(err, "Failed to update CR status %v")
-				return ctrl.Result{}, err
-			}
-
-			r.Log.Info(fmt.Sprintf("OpenStackProvisionServer %s status' LocalImageURL updated: %s", instance.Name, instance.Status.LocalImageURL))
+			common.LogForObject(r, fmt.Sprintf("OpenStackProvisionServer LocalImageURL changed: %s", instance.Status.LocalImageURL), instance)
 		}
 
 		// Now check the associated pod's status
 		podList, err := r.Kclient.CoreV1().Pods(instance.Namespace).List(context.TODO(), metav1.ListOptions{
 			LabelSelector: fmt.Sprintf("deployment=%s-provisionserver-deployment", instance.Name),
 		})
 
-		if err != nil && !k8s_errors.IsNotFound(err) {
-			actualProvisioningState.State = ospdirectorv1beta1.ProvisionServerCondTypeError
-			actualProvisioningState.Reason = err.Error()
-			_ = r.setProvisioningStatus(instance, actualProvisioningState)
+		if err != nil {
+			if k8s_errors.IsNotFound(err) {
+				cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.ProvisionServerCondTypeWaiting)
+				cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.OpenStackProvisionServerCondReasonListError)
+				cond.Message = "List of pods not found"
+				err = common.WrapErrorForObject(cond.Message, instance, err)
+
+				return ctrl.Result{}, err
```

Comment:
shouldn't it RequeueAfter?..

Reply:
Either that or remove this specific error check entirely.  In hindsight, I'm not even sure the `Pods.List()` func can even return an `IsNotFound` error.  It should always return a list, it just might be empty.  So perhaps the proper thing to do is to check the list length after checking for any error in general.

Reply:
I tested the `List` func over the weekend and found that it does not return a `NotFound` error (in contradistinction to the `Get` func).  So I removed the check for `NotFound` and moved that consideration to a block I added to analyze the returned-list's length.